1.train_models/log_reg_plates_3_3p_5_cp_norm_data/nbconverted/plate_3_3p_5_train_log_reg_scv.py

```diff
@@ -291,12 +288,18 @@ def process_plates(_df):
         preds = logreg.predict(X_val)
         acc += accuracy_score(y_val, preds)
 
+        # Store model data for folds
+        eval_data["plate"].extend(restdf.iloc[val_index]["Metadata_Plate"].tolist())
+        eval_data["datasplit"].extend(["val"] * val_index.shape[0])
+        eval_data["predicted_genotype"].extend(preds.tolist())
+        eval_data["true_genotype"].extend(y_val.tolist())
```

Comment:
Consider using the `numpy` arrays without converting to Python lists through [`np.concatenate()`](https://numpy.org/doc/stable/reference/generated/numpy.concatenate.html). It also might be beneficial to consider using [`pd.concat()`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.concat.html) (creating dataframes on each iteration, e.g. `pd.DataFrame({"value_1":np.array, "value_2":np.array})`). The benefit here might be that you are able to make the syntax more declarative and avoid translation from `numpy` to Python objects (dropping the calls to `tolist()`). For larger objects, there could be performance implications here too, but I'm unsure off-hand which would offer a better outcome. Mostly my suggestion would be to avoid data translation if you can avoid it.

Reply:
I've thought about this, I may have to do some benchmarking at some point to determine which is faster. I think appending to lists is faster, but I'm not sure about the data translation piece.